### PR TITLE
#3: added test and documentation for resultSets matcher

### LINKED ISSUE #3: Add unit tests and enable sonar

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #3

### CODE REVIEW

File: src/main/java/com/exasol/matcher/ResultSetMatcher.java [hunk 1]
```diff
@@ -51,45 +49,76 @@ protected boolean matchesSafely(final ResultSet actualResultSet) {
...
-                this.errorMessage.append("Expected and actual result sets have different number of rows.\n");
+            actualNext = actualResultSet.next();
+            this.rowCounter++;
+            if (!bothRowsExist(actualResultSet, expectedNext, actualNext))
                 return false;
-            }
             if (expectedNext && !doesRowMatch(actualResultSet, expectedColumnCount)) {
-                this.errorMessage.append(", row ").append(rowCounter).append(")\n");
                 return false;
             }
```
Comment (redcatbear) [inline src/main/java/com/exasol/matcher/ResultSetMatcher.java:76]: This code makes it very hard to read whether the actual row count will be calculated correctly. The code for that hides in a completely different place: `bothRowsExist()` calls `getRowCounter()`.

File: src/main/java/com/exasol/matcher/ResultSetMatcher.java [hunk 2]
```diff
@@ -51,45 +49,76 @@ protected boolean matchesSafely(final ResultSet actualResultSet) {
...
+        if (expectedColumnCount != actualColumnCount) {
+            this.expectedDescription = "ResultSet with <" + expectedColumnCount + "> column(s)";
+            this.actualDescription = "ResultSet with <" + actualColumnCount + "> column(s)";
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    private boolean bothRowsExist(final ResultSet actualResultSet, final boolean expectedNext, final boolean actualNext)
```
Comment (redcatbear) [inline src/main/java/com/exasol/matcher/ResultSetMatcher.java:91]: Call it `doBothRowsExist` please.

File: src/test/java/com/exasol/matcher/ResultSetMatcherTest.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        // Derby doesn't support two opened result sets on one statement, so we create one more statement.
+        final Statement statement2 = this.connection.createStatement();
+        final ResultSet actual = statement2.executeQuery("SELECT * FROM COL_COUNT_MISMATCH_2");
+        final AssertionError error = assertThrows(AssertionError.class,
+                () -> assertThat(actual, matchesResultSet(expected)));
+        assertThat(error.getMessage(), containsString("Expected: ResultSet with <2> column(s)\n" //
+                + "     but: ResultSet with <1> column(s)"));
+    }
+
+    @Test
```
Comment (redcatbear) [inline src/test/java/com/exasol/matcher/ResultSetMatcherTest.java:51]: Please also test the other way round with more rows in the actual table.

File: src/main/java/com/exasol/matcher/ResultSetMatcher.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/matcher/ResultSetMatcher.java:76]: What can I do to improve it?

File: src/main/java/com/exasol/matcher/ResultSetMatcher.java (hunk 2 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/matcher/ResultSetMatcher.java:91]: done

File: src/test/java/com/exasol/matcher/ResultSetMatcherTest.java (hunk 3 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/matcher/ResultSetMatcherTest.java:51]: done

**Pull request merged**